llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5690,6 +5690,38 @@ Value *SPIRVToLLVM::transSPIRVImageAtomicOpFromInst(SPIRVInstruction *bi, BasicB
   return result;
 }
 
+// =============================================================================
+// Helper function for handling converting sampler select ladder
+template <typename F>
+Value *SPIRVToLLVM::ConvertingSamplerSelectLadderHelper(Value *result, Value *convertingSamplerIdx, F &lambda) {
```

Comment:
Function templates definitions must generally be available to use these functions -- that's why it's very unusual to see function template definition in `.cpp` files instead of headers. I can see that in this case it'd be not very desirable to put all the implementation details there.
Can we avoid using templates altogether here by taking the lambda as `std::function<Value *(Value *)>`?
Or alternatively, can you make it a static out-of-class function template and take `m_convertingSamplers` as a function argument?

Reply:
I prefer the `std::function<Value * (Valid *)>` option. After staring at some disassembly, it seems to result in code that isn't great, but it's able to elide memory allocations, so it's pretty decent.

Reply:
Use std::function now.

Reply:
Thanks!